File: tests/nodejs/cases/case_229_interface_sink_bridge_03.js
/**
 * Task case 229: interface_sink_bridge_03 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case229InterfaceSinkBridge03(input) {
  const taskName = 'interface_sink_bridge_03';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const sink = { put(obj) { retainedAudit.push(obj); } };
  // ESCAPE: object method bridge to retained sink.
  sink.put(payload);
  return 'ok';
}

module.exports = { case229InterfaceSinkBridge03 };

File: tests/python/cases/case_229_interface_sink_bridge_03.py
"""
Task case 229: interface_sink_bridge_03 deep stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_229_interface_sink_bridge_03(input_data):
    task_name = 'interface_sink_bridge_03'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'extreme',
        'stage': 'stress',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    class Sink:
        def put(self, obj):
            RETAINED_AUDIT.append(obj)
    # ESCAPE: interface-like method bridge stores payload globally.
    Sink().put(payload)
    return 'ok'
